datafusion/physical-expr/src/equivalence.rs

```diff
@@ -115,6 +113,53 @@ impl<T: Eq + Hash + Clone> EquivalenceProperties<T> {
     }
 }
 
+/// Remove duplicates inside the `in_data` vector, returned vector would consist of unique entries
+fn deduplicate_vector<T: PartialEq>(in_data: Vec<T>) -> Vec<T> {
+    let mut result = vec![];
+    for elem in in_data {
+        if !result.contains(&elem) {
+            result.push(elem);
+        }
+    }
+    result
+}
+
+/// Find the position of `entry` inside `in_data`, if `entry` is not found return `None`.
+fn get_entry_position<T: PartialEq>(in_data: &[T], entry: &T) -> Option<usize> {
+    in_data.iter().position(|item| item.eq(entry))
+}
+
+/// Remove `entry` for the `in_data`, returns `true` if removal is successful (e.g `entry` is indeed in the `in_data`)
+/// Otherwise return `false`
+fn remove_from_vec<T: PartialEq>(in_data: &mut Vec<T>, entry: &T) -> bool {
```

Comment:
Perhaps a more idiomatic way would be for this function to return `Option<T>` (which is what `Some(in_data.remove())` returns )

That might allow you to avoid some of the other changes to `remove` later

Reply:
Since, we remove by giving element inside the vector. We already have removed element. If we return `Option<T>` the value inside `Option` will be `entry` argument to the function. Hence this function is more akin to `HashSet` `remove`.  Also inside `remove` function we are interested in whether removal was successful, in this case we need to introduce `is_some` checks inside `remove` function.
Hence I think, current API is more clear, However, if it is misleading, or counter intuitive I can implement as your suggestion.

Reply:
makes sense -- thank you for the response